FireboltNETSDK/Client/FireboltCommand.cs

```diff
@@ -187,41 +218,15 @@ public string GetParamQuery(string commandText)
                     }
                     else if (item.Value is bool)
                     {
-                        if ((bool)item.Value)
-                            verifyParameters = "1";
-                        else
-                            verifyParameters = "0";
+                        verifyParameters = (bool)item.Value ? "1" : "0";
                     }
                     else if (item.Value is IList && item.Value.GetType().IsGenericType)
                     {
                         throw new FireboltException("Array query parameters are not supported yet.");
                     }
-                    else if (item.Value is int || item.Value is long || item.Value is double || item.Value is float ||
-                             item.Value is decimal)
+                    else if (item.Value is IConvertible)
                     {
-                        switch (item.Value.GetType().Name)
-                        {
-                            case "Decimal":
-                                var decValue = (decimal)item.Value;
-                                verifyParameters = decValue.ToString().Replace(',', '.');
-                                break;
-                            case "Double":
-                                var doubleValue = (double)item.Value;
-                                verifyParameters = doubleValue.ToString().Replace(',', '.');
-                                break;
-                            case "Single":
-                                var floatValue = (float)item.Value;
-                                verifyParameters = floatValue.ToString().Replace(',', '.');
-                                break;
-                            case "Int32":
-                                var intValue = (int)item.Value;
-                                verifyParameters = intValue.ToString();
-                                break;
-                            case "Int64":
-                                var longValue = (long)item.Value;
-                                verifyParameters = longValue.ToString();
-                                break;
-                        }
+                        verifyParameters = ((IConvertible)item.Value).ToString(new CultureInfo("en-US", false));
                     }
```

Comment:
Can you explain what does this do and why is it being removed please?

Reply:
You are right, this code is not obvious, so I added comment to code. If you want to see the reason see the full page of code that was in previous version and is replaced by this single line.